packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -103,6 +123,49 @@ export function unstable_httpSubscriptionLink<
           >({
             from: eventSource,
             deserialize: transformer.output.deserialize,
+            tryHandleError: async (ev) => {
+              if (
+                'status' in ev &&
+                typeof ev.status === 'number' &&
+                typeof opts.shouldRecreateOnError === 'function' &&
+                eventSource
+              ) {
+                let recreateOnErrorOpts: RecreateOnErrorOpt = {
+                  type: 'raw',
+                  event: ev,
+                };
+                if ('status' in ev && typeof ev.status === 'number') {
```

Comment:
Avoid nested `if`s like this, this code is very hard to follow

do an early return or something instead